classes/publication/DAO.php

```diff
@@ -55,4 +64,81 @@ public function fromRow(object $primaryRow): Publication
+    {
+        $q = DB::table('publications', 'p')
+            ->join('submissions AS s', 's.submission_id', '=', 'p.submission_id')
+            ->leftJoin(
+                'publications AS p2',
+                fn (JoinClause $j) => $j->on('p2.submission_id', '=', 'p.submission_id')
+                    ->on('p2.version_stage', '=', 'p.version_stage')
+                    ->on('p2.version_major', '=', 'p.version_major')
+                    ->on('p.version_minor', '<', 'p2.version_minor')
+                    ->where('p2.status', '=', Submission::STATUS_PUBLISHED)
+            )
+            ->when(
+                $pubIdType != null,
+                fn (Builder $q) => $q->leftJoin('publication_settings AS pspidt', 'p.publication_id', '=', 'pspidt.publication_id')
+            )
+            ->when($title != null, fn (Builder $q) => $q->leftJoin('publication_settings AS pst', 'p.publication_id', '=', 'pst.publication_id'))
+            ->when(
+                $author != null,
+                fn (Builder $q) => $q->leftJoin('authors AS au', 'p.publication_id', '=', 'au.publication_id')
+                    ->leftJoin(
+                        'author_settings AS asgs',
+                        fn (JoinClause $j) => $j->on('asgs.author_id', '=', 'au.author_id')
+                            ->where('asgs.setting_name', '=', Identity::IDENTITY_SETTING_GIVENNAME)
+                    )
+                    ->leftJoin(
+                        'author_settings AS asfs',
+                        fn (JoinClause $j) => $j->on('asfs.author_id', '=', 'au.author_id')
+                            ->where('asfs.setting_name', '=', Identity::IDENTITY_SETTING_FAMILYNAME)
+                    )
+            )
+            ->when(
+                $settingName,
+                fn (Builder $q) => $q->leftJoin(
+                    'publication_settings AS pss',
+                    fn (JoinClause $j) => $j->on('p.publication_id', '=', 'pss.publication_id')
+                        ->where('pss.setting_name', '=', $settingName)
+                )
+            )
+            ->where('s.status', '=', Submission::STATUS_PUBLISHED)
+            ->where('s.context_id', '=', $contextId)
+            ->where('p.version_stage', '=', VersionStage::VERSION_OF_RECORD)
+            ->where('p.status', '=', Publication::STATUS_PUBLISHED)
+            ->whereNull('p2.publication_id')
+            ->when($pubIdType != null, fn (Builder $q) => $q->where('pspidt.setting_name', '=', "pub-id::{$pubIdType}")->whereNotNull('pspidt.setting_value'))
+            ->when($title != null, fn (Builder $q) => $q->where('pst.setting_name', '=', 'title')->where('pst.setting_value', 'LIKE', "%{$title}%"))
+            ->when($author != null, fn (Builder $q) => $q->whereRaw("CONCAT(COALESCE(asgs.setting_value, ''), ' ', COALESCE(asfs.setting_value, '')) LIKE ?", [$author]))
```

Comment:
Could adjust this to allow for only last name/first name searches (e.g. %author% or something similar)

Reply:
if we allow this only for last and first name, should we change it in the UI as well to be "Author name" instead of "Author"?

Reply:
What do you exactly mean? To allow searches with only partial match (%author%)?

Reply:
I was thinking for example if you were to search for only the last name, to handle that case.

Reply:
Ah yes, good point!